### PR TITLE
Added Algo/longest_increasing_subsequence [C++]

### LINKED ISSUE #488: Algo/LongestIncreasingSubsequence
I would like to add O(nlogn) implementation of Longest Increasing Subsequence in an array. 
#### This issue is about [Algo/DS Name](link to resource for the Algo/DS).
http://codeforces.com/blog/entry/12274
- [x] Issue Title is in form `Algo/DS Name [Language]`.
- [x] I searched or browsed the repo’s other issues to ensure this is not a duplicate.
- [x] This Algo/DS is worth adding to this repository.
- [x] Assign this issue to me (I want to work on this).
	- [ ] I ensure that I am not already assigned to 2 or more issues.
- [x] I will send my PR only after I'm assigned to this issue by any maintainer.
- [x] I understand that I will be unassigned from this issue if I fail to send a PR for this within a week.

### PR DESCRIPTION
**Pull request by manish1997**
Fixes #488 

#### By submitting this pull request I confirm I've read and complied with the below declarations.

- [x] I have read the [Contribution guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md) and I am confident that my PR reflects them.
- [x] I have followed the [coding guidelines](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#cs) for this project.
- [x] My code follows the [skeleton code structure](https://github.com/iiitv/algos/blob/master/CONTRIBUTING.md#sample).
- [x] This pull request has a descriptive title. For example, `Added {Algorithm/DS name} [{Language}]`, not `Update README.md` or `Added new code`.
- [x] This pull request will be closed if I fail to update it even once in a continuous time span of 7 days.

### CODE REVIEW

**Comment by singhpratyush** [comment]
Hi! Thanks for showing interest in contributing to this repository.

Please make sure that you have ticked the points mentioned in PR description correctly.

Thanks again!

**Comment by manish1997** [comment]
@aashutoshrathi @singhpratyush

**Comment by singhpratyush** [comment]
Please add an entry to README.

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 1]
```diff
@@ -0,0 +1,29 @@
+//Time: O(n*log(n))
+#include <bits/stdc++.h>
```
Comment (mohitkyadav) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Only include used header files.

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+				if(S[mid]>=A[i])  end=mid-1;
+				else start=mid+1;
+			}
+			S[end+1]=A[i];
+		}
+	}
+	return S.size();
+}
+int main(){
+	int n; cin >> n; // size of array for which LIS needs to be found
```
Comment (mohitkyadav) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Please put comments above the respective lines.
i.e.
```
// Comment
int n;
```

**Comment by Monal5031** [comment]
@manish1997 Please make the requested changes.

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 3]
```diff
@@ -0,0 +1,35 @@
+//Time: O(n*log(n))
+#include <iostream>
+#include <vector>
+#include <cstring>
+#include <cmath>
+#include <climits>
+using namespace std;
+
+int LIS ( vector<int> A ) {
```
Comment (singhpratyush) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Please use the complete name here. Also, do not add extra spaces.

```cpp
int my_method(vector<int> a) {
     ...
}
```

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+	if ( A.size() <= 1 ) return A.size();
```
Comment (singhpratyush) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Please add line break after conditionals.

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+	vector< int > S;
```
Comment (singhpratyush) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: `vector<int>`

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+	S.push_back( A[0] );
+	for ( int i = 1; i < A.size(); i++ ) {
```
Comment (singhpratyush) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: `for (...) {`

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+				int mid = start+(end-start)/2;
+				if( S[mid] >= A[i] )  end=mid-1;
+				else start = mid+1;
+			}
+			S[end+1] = A[i];
+		}
+	}
+	return S.size();
+}
+int main() {
```
Comment (singhpratyush) [inline longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Please add a line break before.

File: longest_increasing_subsequence/longestIncreasingSubsequence.cpp (hunk 6 above)
Comment (manish1997) [reply longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: @singhpratyush what's the problem here?

Comment (singhpratyush) [reply longest_increasing_subsequence/longestIncreasingSubsequence.cpp]: Do not add extra spaces like 

```
for ( int...
     ^
```

**Comment by Monal5031** [comment]
Closing due to inactivity.

**Pull request closed without merging**